### PR TITLE
JAMES-2078 Disable  dynamic mappings for Mailbox indexing

### PR DESCRIPTION
**Pull request by chibenwa**
This can be achieved through the use of nested objects for headers.

Adoption requires a reIndexing.

### CODE REVIEW

**Comment by chibenwa** [comment]
```

[e33fb743d9e404f5e729564b8e43fe916f22b80f] [ERROR]   IndexableMessageTest.textShouldContainsBccWhenBcc:198 
[e33fb743d9e404f5e729564b8e43fe916f22b80f] Expecting:
[e33fb743d9e404f5e729564b8e43fe916f22b80f]  <"First bcc user@james.org Second bcc user2@james.org">
[e33fb743d9e404f5e729564b8e43fe916f22b80f] to be equal to:
[e33fb743d9e404f5e729564b8e43fe916f22b80f]  <"Second bcc user2@james.org First bcc user@james.org">
[e33fb743d9e404f5e729564b8e43fe916f22b80f] but was not.
[e33fb743d9e404f5e729564b8e43fe916f22b80f] [ERROR]   IndexableMessageTest.textShouldContainsFromWhenFrom:123 
[e33fb743d9e404f5e729564b8e43fe916f22b80f] Expecting:
[e33fb743d9e404f5e729564b8e43fe916f22b80f]  <"First user user@james.org Second user user2@james.org">
[e33fb743d9e404f5e729564b8e43fe916f22b80f] to be equal to:
[e33fb743d9e404f5e729564b8e43fe916f22b80f]  <"Second user user2@james.org First user user@james.org">
```

**Comment by chibenwa** [comment]
test this please

File: mailbox/store/src/test/resources/eml/pgpSignedMail.json [hunk 1]
```diff
@@ -10,121 +10,136 @@
...
+    "name": "received",
+    "value": "from mx2.minet.net ([IPv6:::ffff:192.168.102.26])\tby localhost (spam.minet.net [::ffff:192.168.102.97]) (amavisd-new, port 10024)\twith ESMTP id cF6J0AAnzsTX for <benwa@minet.net>;\tWed,  3 Jun 2015 19:20:02 +0000 (UTC)"
+  }, {
+    "name": "received-spf",
+    "value": "None (no SPF record) identity=mailfrom; client-ip=82.195.75.100; helo=bendel.debian.org; envelope-from=bounce-debian-security-announce=benwa=minet.net@lists.debian.org; receiver=benwa@minet.net"
+  }, {
+    "name": "received",
+    "value": "from bendel.debian.org (bendel.debian.org [82.195.75.100])\t(using TLSv1.2 with cipher AECDH-AES256-SHA (256/256 bits))\t(No client certificate requested)\tby mx2.minet.net (Postfix) with ESMTPS id 7F69FA27D98\tfor <benwa@minet.net>; Wed,  3 Jun 2015 21:20:01 +0200 (CEST)"
+  }, {
+    "name": "received",
```
Comment (rouazana) [inline mailbox/store/src/test/resources/eml/pgpSignedMail.json]: it's not an issue to have several headers with the same name?

Comment (chibenwa) [reply mailbox/store/src/test/resources/eml/pgpSignedMail.json]: No

**Comment by chibenwa** [comment]
Just rebased on last changes of JAMES-2079

**Comment by chibenwa** [comment]
```
[d6fab3fe08ddcd6d53fe047155c9cc745a5e8ab6] [ERROR]   MemorySetMessagesMethodTest>SetMessagesMethodTest.setMessagesShouldSetTheSeenKeywordOnMessageInSentMailbox:5498 1 expectation failed.
[d6fab3fe08ddcd6d53fe047155c9cc745a5e8ab6] JSON path [1][1].list[0].keywords.$Seen doesn't match.
[d6fab3fe08ddcd6d53fe047155c9cc745a5e8ab6] Expected: <true>
[d6fab3fe08ddcd6d53fe047155c9cc745a5e8ab6]   Actual: null
```

test this please

**Comment by chibenwa** [comment]
Regarding performance:

 - Default message list performance is unimpacted
 - We noticed a 4% performance improvment upon indexing throughtput
 - We noticed a 7% increase regarding space per message

I believe we should adopt this proposal.

**Comment by chibenwa** [comment]
Slightly more, yes

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**